crates/sui-rosetta/docker/Dockerfile

```diff
@@ -0,0 +1,57 @@
+FROM ubuntu:latest AS chef
+WORKDIR sui
+ARG GIT_REVISION
+ENV GIT_REVISION=$GIT_REVISION
+RUN apt-get update && apt-get install -y build-essential libssl-dev pkg-config curl cmake clang
+RUN curl https://sh.rustup.rs -sSf | bash -s -- -y
+ENV PATH="/root/.cargo/bin:${PATH}"
+
+# Plan out the 3rd-party dependencies that need to be built.
+#
+# This is done by:
+#   1. Copy in Cargo.toml, Cargo.lock, and the workspace-hack crate
+#   2. Removing all workspace crates, other than the workpsace-hack
+#      crate, from the workspace Cargo.toml file.
+#   3. Update the lockfile in order to reflect the changes to the
+#      root Cargo.toml file.
+FROM chef AS planner
+COPY Cargo.toml Cargo.lock ./
+COPY crates/workspace-hack crates/workspace-hack
+RUN sed -i '/crates\/workspace-hack/b; /crates/d' Cargo.toml \
+    && cargo metadata -q >/dev/null
+
+# Build and cache all dependencies.
+#
+# In a fresh layer, copy in the "plan" generated by the planner
+# and run `cargo build` in order to create a caching Docker layer
+# with all dependencies built.
+FROM chef AS builder
```

Comment:
I will be changing this to build from git clone instead of copy from source after this is merged, it is required by Rosetta.
https://www.rosetta-api.org/docs/node_deployment.html#dockerfile-expectations